### PR TITLE
add post release tests for manifests archive contents

### PR DESCRIPTION
**Pull request by radTuti**
## Description

This adds tests to verify the manifests of the archive included in a release.

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: hack/postrelease/test_manifests_archives.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    "calicoctl.yaml",
+    "calicoctl-etcd.yaml",
+    "canal.yaml",
+    "canal-etcd.yaml",
+    "tigera-operator.yaml",
+    "custom-resources.yaml",
+]
+
+
+def test_manifest_archive_on_release_page():
```
Comment (lwr20) [inline hack/postrelease/test_manifests_archives.py:32]: This works fine as-is, how hard would it be to make this a parameterised test (like in `test_tags_tars`).  That way the output is more explicit about exactly which files have been found/not found in the archive.

**Review by lwr20 — APPROVED** [review]
Approved, one suggestion for improving.

File: hack/postrelease/test_manifests_archives.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import requests
+import tarfile
+
+from versions import RELEASE_VERSION
+
+urls = {
+    "docs": "https://github.com/projectcalico/calico/releases/download/{v}/release-{v}.tgz".format(
```
Comment (caseydavenport) [inline hack/postrelease/test_manifests_archives.py:7]: These are both the same URL - is there a reason we need both?

Comment (radTuti) [reply hack/postrelease/test_manifests_archives.py]: oversight

File: hack/postrelease/test_manifests_archives.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+
+def teardown_archive():
+    file.close()
+
+
+@with_setup(setup=setup_archive, teardown=teardown_archive)
+def test_manifest_present():
+    for manifest in manifest_list:
+        yield check_manifest_present, manifest
```
Comment (caseydavenport) [inline hack/postrelease/test_manifests_archives.py:41]: I don't think a `yield` is the right execution method here - there's never a situation where we're wanting to pause execution of check_manifest_present and then start where we left off, right?

Probably better to just use a function call - `check_manifest_present(manifest)`

Comment (radTuti) [reply hack/postrelease/test_manifests_archives.py:41]: i'm using nosetest[ test generators](https://nose.readthedocs.io/en/latest/writing_tests.html#test-generators) which creates individual tests for each manifest - more explict with failure

a function call gives this result summary

```sh
test_manifests_archives.test_manifest_present ... ok (23.9096s)
```

comapred to a yieded tests

```sh
test_manifests_archives.test_manifest_present('calico.yaml',) ... ok (23.4394s)
test_manifests_archives.test_manifest_present('calico-etcd.yaml',) ... ok (0.0001s)
test_manifests_archives.test_manifest_present('calico-bpf.yaml',) ... ok (0.0003s)
test_manifests_archives.test_manifest_present('calico-typha.yaml',) ... ok (0.0002s)
test_manifests_archives.test_manifest_present('calico-vxlan.yaml',) ... ok (0.0002s)
test_manifests_archives.test_manifest_present('calico-windows-bgp.yaml',) ... ok (0.0003s)
test_manifests_archives.test_manifest_present('calico-windows-vxlan.yaml',) ... ok (0.0002s)
test_manifests_archives.test_manifest_present('calicoctl.yaml',) ... ok (0.0002s)
test_manifests_archives.test_manifest_present('calicoctl-etcd.yaml',) ... ok (0.0002s)
test_manifests_archives.test_manifest_present('canal.yaml',) ... ok (0.0002s)
test_manifests_archives.test_manifest_present('canal-etcd.yaml',) ... ok (0.0002s)
test_manifests_archives.test_manifest_present('tigera-operator.yaml',) ... ok (0.0002s)
test_manifests_archives.test_manifest_present('custom-resources.yaml',) ... ok (0.0002s)
```

Comment (caseydavenport) [reply hack/postrelease/test_manifests_archives.py:41]: Ahhh, gotcha. Unfortunate that we still have Python tests hanging around but maybe one day we'll switch these all over to go. Good for now.

**Pull request merged**